extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/HadoopFileSourceFactory.java

```diff
@@ -0,0 +1,237 @@
+import com.hazelcast.jet.pipeline.file.JsonFileFormat;
+import com.hazelcast.jet.pipeline.file.LinesTextFileFormat;
+import com.hazelcast.jet.pipeline.file.ParquetFileFormat;
+import com.hazelcast.jet.pipeline.file.RawBytesFileFormat;
+import com.hazelcast.jet.pipeline.file.TextFileFormat;
+import com.hazelcast.jet.pipeline.file.impl.FileSourceFactory;
+import org.apache.avro.Schema;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.reflect.ReflectData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
+import org.apache.parquet.avro.AvroParquetInputFormat;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static com.hazelcast.jet.hadoop.impl.CsvInputFormat.CSV_INPUT_FORMAT_BEAN_CLASS;
+import static com.hazelcast.jet.hadoop.impl.JsonInputFormat.JSON_INPUT_FORMAT_BEAN_CLASS;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Hadoop-based implementation of {@link FileSourceFactory}.
+ */
+public class HadoopFileSourceFactory implements FileSourceFactory {
+
+    private final Map<String, JobConfigurer> configs;
+
+    /**
+     * Creates the HadoopSourceFactory.
+     */
+    public HadoopFileSourceFactory() {
+        configs = new HashMap<>();
+
+        configs.put(AvroFileFormat.FORMAT_AVRO, new AvroFormatJobConfigurer());
+        configs.put(CsvFileFormat.FORMAT_CSV, new CsvFormatJobConfigurer());
+        configs.put(JsonFileFormat.FORMAT_JSONL, new JsonFormatJobConfigurer());
+        configs.put(LinesTextFileFormat.FORMAT_LINES, new LineTextJobConfigurer());
+        configs.put(ParquetFileFormat.FORMAT_PARQUET, new ParquetFormatJobConfigurer());
+        configs.put(RawBytesFileFormat.FORMAT_BIN, new RawBytesFormatJobConfigurer());
+        configs.put(TextFileFormat.FORMAT_TXT, new TextJobConfigurer());
+    }
+
+    @Nonnull @Override
+    @SuppressWarnings("unchecked")
+    public <T> BatchSource<T> create(@Nonnull FileSourceBuilder<T> builder) {
+
+        try {
+            Job job = Job.getInstance();
+
+            Configuration configuration = job.getConfiguration();
+            for (Entry<String, String> option : builder.options().entrySet()) {
+                configuration.set(option.getKey(), option.getValue());
+            }
+
+            FileInputFormat.addInputPath(job, new Path(builder.path()));
+
+            FileFormat<T> fileFormat = requireNonNull(builder.format());
+            JobConfigurer configurer = configs.get(fileFormat.format());
```

Comment:
we should do a null check here for custom file format

Reply:
Added.